usaspending_api/broker/management/commands/load_fpds_from_broker.py

```diff
@@ -0,0 +1,139 @@
+            lines = [line.decode("utf-8") for line in lines]
+            if len(lines) == 0:
+                break
+            yield lines
+
+    def load_fpds_from_file(self, file_path):
+        with RetrieveFileFromUri(file_path).get_file_object() as file:
+            for next_batch in self.next_file_batch_generator(file):
+                id_list = [int(re.search(r"\d+", x).group()) for x in next_batch]
+                logger.info(
+                    "Loading next batch from provided file (size: {}, ids {}-{})...".format(
+                        len(id_list), id_list[0], id_list[-1]
+                    )
+                )
+                self.modified_award_ids.extend(run_fpds_load(id_list))
+
+    def add_arguments(self, parser):
+        mutually_exclusive_group = parser.add_mutually_exclusive_group()
+
+        mutually_exclusive_group.add_argument(
+            "--ids",
+            nargs="+",
+            type=int,
+            help="(OPTIONAL) detached_award_procurement_ids of FPDS transactions to load/reload from Broker",
+        )
+        mutually_exclusive_group.add_argument(
+            "--date",
+            dest="date",
+            type=datetime_command_line_argument_type(naive=True),  # Broker date/times are naive.
+            help="Load or Reload all FPDS records from the provided date to the current time. YYYY-MM-DD format",
+        )
+        mutually_exclusive_group.add_argument(
+            "--since-last-load",
+            action="store_true",
+            help="Equivalent to loading from date, but date is drawn from last update date recorded in DB",
+        )
+        mutually_exclusive_group.add_argument(
+            "--file",
+            metavar="FILEPATH",
+            type=str,
+            help="A file containing only transaction IDs (detached_award_procurement_id) "
+            "to reload, one ID per line. Nonexistent IDs will be ignored.",
+        )
+        mutually_exclusive_group.add_argument(
+            "--reload-all",
+            action="store_true",
+            help="Script will load or reload all FPDS records in broker database, from all time",
+        )
+
+    def handle(self, *args, **options):
+
+        # loads can take a while, so we record last updated date from the start of all transactions
+        last_update_time = datetime.now(timezone.utc)
+
+        if options["reload_all"]:
+            self.load_fpds_from_date(None)
+
+        elif options["date"]:
+            self.load_fpds_from_date(options["date"])
+
+        elif options["ids"]:
+            self.modified_award_ids.extend(run_fpds_load(options["ids"]))
+
+        elif options["file"]:
+            self.load_fpds_from_file(options["file"])
+
+        elif options["since_last_load"]:
+            # minus one day to account for any submissions that are still entering the system when the loader is run
+            self.load_fpds_from_date(get_last_load_date("fpds") - timedelta(days=1))
```

Comment:
Why are you adding 1 day to the last load date? This is very different than the behavior of the current fpds loader script

Reply:
Shouldn't this be subtracting one day? My testing showed that it pulls transactions from one day before the last load date in the DB, in order to account for very slow transactions

Reply:
The current FPDS loader does not do this

Reply:
Missed the update on Kirk's review. Thanks for the spot